src/ProjectModule/Components/ProjectForm/ProjectForm.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useState } from 'react'
+import styles from './ProjectForm.module.css'
+import { useNavigate, useParams } from 'react-router-dom'
+import { useForm } from 'react-hook-form';
+import axios from 'axios';
+import { baseUrl } from '../../../Constants/Components/Urls';
+import { toast } from 'react-toastify';
+import Loader from "../../../SharedMoudule/Components/Loading/Loading";
+
+
+export default function ProjectForm() {
+    const navigate=useNavigate();
+    const {register,handleSubmit,formState:{errors}}=useForm();
+    const token=localStorage.getItem('userToken');
+    const {action}=useParams();
+    const [currentProject,setCurrentProject]=useState(null);
+    const curruntProjectId=localStorage.getItem('curruntProjectId');
+    const [isLoading, setIsLoading] = useState(false);
+
+    const onSubmitAdd= async(data:any)=>{
```

Comment:
The code of the add and update are almost the same, the only thing that changes is the method and and URL path and the messages displayed, we can change this dynamically and make them in one function.